packages/circuit-ui/components/DateInput/components/Dialog.tsx

```diff
@@ -45,7 +46,21 @@ export const Dialog = forwardRef<HTMLDialogElement, DialogProps>(
     const zIndex = useStackContext();
     const dialogRef = useRef<HTMLDialogElement>(null);
 
-    useClickOutside(dialogRef, onClose, open);
+    // the last focused element, used to restore focus when the dialog is closed
+    const lastFocusedElementRef = useRef<Element | null>(null);
+
+    const handleClickOutside = useCallback(
+      // store it as the last focused element
+      (event: Event) => {
+        if (event.target instanceof HTMLElement) {
+          lastFocusedElementRef.current = event.target;
+        }
```

Comment:
I see what you did here 👀 

It's clever, but took me a minute to understand 😅 Please add a code comment to explain why the last focused element has to be overwritten with the clicked element.